sdks/python/apache_beam/io/external/snowflake.py

```diff
@@ -0,0 +1,144 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+from __future__ import absolute_import
+
+import typing
+
+from past.builtins import unicode
+
+import apache_beam as beam
+from apache_beam.transforms.external import BeamJarExpansionService
+from apache_beam.transforms.external import ExternalTransform
+from apache_beam.transforms.external import NamedTupleBasedPayloadBuilder
+
+__all__ = ['ReadFromSnowflake']
+
+
+def default_io_expansion_service():
+  return BeamJarExpansionService('sdks:java:io:expansion-service:shadowJar')
```

Comment:
You have to either add snowflake jars to "io:expansion-service" or expose a new Snowflake shadow jar that this wrapper can use (for expansion and for staging for runtime). What will be the size diff of "sdks:java:io:expansion-service:shadowJar" after including Snowflake ? If that is large we should go for the second option since this jar is shared by a bunch of cross-language wrappers.

Reply:
After adding snowflake to "io:expansion-service", jar size increased from  52,4 MB to 117,4 MB, so I think we should go with the second option, can you give me some hints how to proceed with it?

Reply:
One solution might be to add a new expansion service jar just for Snowflake but I'm not sure if we want to keep adding expansion service jars for various IOs. On the other hand I don't think basically doubling the size of existing IO expansion service jar is acceptable since that will affect the staging/pipeline construction time for users of all Java external sources.

@robertwb for thoughts here.

Reply:
Yes, I think going with a separate jar makes sense given its size. 

To do this, create a rule for a fat jar in `sdks/java/io/snowflake/build.gradle` that depends on both `sdks:java:expansion-service` and `sdks:java:io:snowflake`. (I'm not an expert on how the java nature creates fat jars; perhaps the simplest would be to put this in a new package `sdks/java/io/snowflake/expansion-service` whose sole contents would be exactly the same as `sdks/java/io/expansion-service/build.gradle` except depend on snowflake rather than kafka and jdbc.

Reply:
Thanks @robertwb. I added it